From: Simon McVittie <smcv@debian.org>
Date: Mon, 19 Nov 2018 15:26:31 +0000
Subject: test-doc-portal: Skip all tests if FUSE isn't supported

Distribution autobuilders sometimes use a hardened OS configuration that
does not allow module loading or FUSE; in particular, this is the case
in Debian. In this environment, global setup does not complete,
@documents is NULL and this test would segfault. Skip the test and rely
on installed-tests to cover it instead.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/xdg-desktop-portal/pull/263
---
 tests/test-doc-portal.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/test-doc-portal.c b/tests/test-doc-portal.c
index f2c0f22..10b9b5d 100644
--- a/tests/test-doc-portal.c
+++ b/tests/test-doc-portal.c
@@ -858,6 +858,12 @@ global_teardown (void)
 static void
 test_version (void)
 {
+  if (!have_fuse)
+    {
+      g_test_skip ("this test requires FUSE");
+      return;
+    }
+
   g_assert_cmpint (xdp_dbus_documents_get_version (documents), ==, 3);  
 }
 
